Task 18

PaolkaT opened the pull request.

PaolkaT committed:
TASK-18: Publish Brench, create PetRepo, PetDTO

PaolkaT committed:
TASK 18: Create endpoinds

1) Add new pet-card
2) Delete pet-card
3) Check pet page

PaolkaT committed:
TASK 18: Fix infinite recursion

BrineFast commented on be/src/main/java/com/savethepet/controller/PetPageController.java:
```diff
@@ -0,0 +1,112 @@
+
+import com.savethepet.model.dto.user.PetInfoChangeDTO;
+import com.savethepet.model.dto.user.PetInfoDTO;
+import com.savethepet.model.dto.user.UserInfoDTO;
```
Useless import

BrineFast commented on be/src/main/java/com/savethepet/controller/PetPageController.java:
```diff
@@ -0,0 +1,112 @@
+import com.savethepet.model.dto.user.PetInfoDTO;
+import com.savethepet.model.dto.user.UserInfoDTO;
+import com.savethepet.model.entity.Pet;
+import com.savethepet.model.entity.User;
```
Useless import

BrineFast commented on be/src/main/java/com/savethepet/controller/PetPageController.java:
```diff
@@ -0,0 +1,112 @@
+    public void addPet(@RequestBody @Valid PetInfoChangeDTO petInfoChangeDTO,
+                       Long user_id) {
+        petPageService.addingPet(petInfoChangeDTO, user_id);
+    }
```
It would be nice to return the entity after creation

BrineFast replied:
And where is the exception

BrineFast commented on be/src/main/java/com/savethepet/controller/PetPageController.java:
```diff
@@ -0,0 +1,112 @@
+    public void deletePet(@ApiIgnore Principal principal, @PathVariable("pet_id") Long id) {
+        petPageService.deletePet(id);
+    }
+}
```
And pls rename `id` to `pet_id`

BrineFast commented on be/src/main/java/com/savethepet/controller/UserPageController.java:
```diff
@@ -1,5 +1,6 @@
 package com.savethepet.controller;
 
+import com.savethepet.model.dto.user.PetInfoDTO;
```
Useless import on this moment

BrineFast commented on be/src/main/java/com/savethepet/controller/UserPageController.java:
```diff
@@ -50,13 +51,31 @@ public UserInfoDTO getUserInfo(@PathVariable("user_id") Long id) {
+
     /**
      * Changes info about user
      *
```
???????????

BrineFast commented on be/src/main/java/com/savethepet/controller/PetPageController.java:
```diff
@@ -0,0 +1,112 @@
+    @PatchMapping("/pet/{pet_id}")
+    public void changePetInfo(@PathVariable("pet_id") Long id, @RequestBody @Valid PetInfoChangeDTO petInfoChangeDTO) {
+        petPageService.updatePetFromDto(petInfoChangeDTO, id);
+    }
```
It would be nice to return the entity after editing

BrineFast commented on be/src/main/java/com/savethepet/model/dto/user/PetInfoChangeDTO.java:
```diff
@@ -0,0 +1,28 @@
+    
+    private PetLocation location;
+
+    private Status status;
```
`@NotBlank`

BrineFast commented on be/src/main/java/com/savethepet/model/entity/Pet.java:
```diff
@@ -1,6 +1,13 @@
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.JsonManagedReference;
+import com.fasterxml.jackson.annotation.ObjectIdGenerator;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
```
Useless

BrineFast commented on be/src/main/java/com/savethepet/model/entity/Pet.java:
```diff
@@ -11,13 +18,27 @@
 @Data
+@AllArgsConstructor
+@NoArgsConstructor
+//@EqualsAndHashCode
```
?????

BrineFast commented on be/src/main/java/com/savethepet/model/entity/Pet.java:
```diff
@@ -1,6 +1,13 @@
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+import lombok.AllArgsConstructor;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
```
Useless

BrineFast commented on be/src/main/java/com/savethepet/model/entity/PetLocation.java:
```diff
@@ -0,0 +1,9 @@
+package com.savethepet.model.entity;
+
+import lombok.Data;
+
+@Data
```
Any comments?

BrineFast commented on be/src/main/java/com/savethepet/model/entity/User.java:
```diff
@@ -1,15 +1,20 @@
 package com.savethepet.model.entity;
 
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
```
Useless imports

BrineFast commented on be/src/main/java/com/savethepet/model/entity/User.java:
```diff
@@ -1,15 +1,20 @@
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
+import org.codehaus.jackson.annotate.JsonBackReference;
+import org.hibernate.annotations.Cascade;
```
Useless imports

BrineFast commented on be/src/main/java/com/savethepet/model/entity/User.java:
```diff
@@ -1,15 +1,20 @@
 import javax.persistence.*;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.Set;
```
Useless imports

BrineFast commented on be/src/main/java/com/savethepet/service/PetPageService.java:
```diff
@@ -0,0 +1,102 @@
+@Service
+public class PetPageService {
+
+    private static final String userWithOauthId = "Pet with id = ";
```
User?

BrineFast commented on be/src/main/java/com/savethepet/service/PetPageService.java:
```diff
@@ -0,0 +1,102 @@
+                new PetNotFoundException("Pet with id = " + id.toString() + notFound)).get
+        return petRepo.findById(id).orElseThrow(() ->
+                new PetNotFoundException("Pet with id = " + id.toString() + notFound));
+    }*/
```
???

BrineFast commented on be/src/main/java/com/savethepet/service/PetPageService.java:
```diff
@@ -0,0 +1,102 @@
+     * @param id
+     * @return
+     */
+    public Pet getPetById(Long id) {
```
Rename to `pet_id`

BrineFast commented on be/src/main/java/com/savethepet/service/PetPageService.java:
```diff
@@ -0,0 +1,102 @@
+        changedPet.setGender(petInfoChangeDTO.getGender());
+        changedPet.setStatus(petInfoChangeDTO.getStatus());
+        petRepo.save(changedPet);
+    }
```
How about adding photos?

BrineFast replied:
And return `changerPet` pls

BrineFast commented on be/src/main/java/com/savethepet/service/PetPageService.java:
```diff
@@ -0,0 +1,102 @@
+        petFromDTO.setLocation(petInfoDTO.getLocation());
+        petFromDTO.setStatus(petInfoDTO.getStatus());
+        petRepo.save(petFromDTO);
+    }
```
Return this entity pls

BrineFast commented on be/src/main/java/com/savethepet/service/PetPageService.java:
```diff
@@ -0,0 +1,102 @@
+        petRepo.findById(id).orElseThrow(() ->
+                new PetNotFoundException("Pet with id = " + id.toString() + notFound));
+        petRepo.deleteById(id);
+    }
```
It is not very good to make 2 requests to the database for deletion

PaolkaT committed:
TASK 18: Add two methods

getPetsByUserId and getPetsFromBreed

PaolkaT committed:
TASK 18: Fix some bugs

onxas committed:
TASK-18: Fixed filter and User Pets

Updated User and Pet Controller and services
Added custom Query to filter Pets vy breed, status and user

PaolkaT committed:
TASK-18: Fix swagger and javadoc

BrineFast commented on be/src/main/java/com/savethepet/controller/PetPageController.java:
```diff
@@ -0,0 +1,124 @@
+    public PetInfoDTO getPetInfo(@PathVariable("pet_id") Long id) {
+        Pet pet = petPageService.getPetById(id);
+        return PetInfoDTO.getDtoFromPet(pet);
+    }
```
That would be better: `PetInfoDTO.getDtoFromPet(petPageService.getPetById(id))`

BrineFast commented on be/src/main/java/com/savethepet/controller/PetPageController.java:
```diff
@@ -0,0 +1,124 @@
+    public PetInfoChangeDTO changePetInfo(@PathVariable("pet_id") Long id, @RequestBody @Valid PetInfoChangeDTO petInfoChangeDTO) {
+        petPageService.updatePetFromDto(petInfoChangeDTO, id);
+        return petInfoChangeDTO;
+    }
```
If you have the `Pet 'class being returned in the service, why can't you make `PetDTO` return and write `return petPageService.updatePetFromDto(petInfoChangeDTO, id)`

BrineFast commented on be/src/main/java/com/savethepet/controller/PetPageController.java:
```diff
@@ -0,0 +1,124 @@
+                                   Long user_id) {
+        petPageService.addingPet(petInfoChangeDTO, user_id);
+        return petInfoChangeDTO;
+    }
```
Another comment about the return of the entity

BrineFast commented on be/src/main/java/com/savethepet/model/dao/PetRepo.java:
```diff
@@ -0,0 +1,25 @@
+
+import com.savethepet.model.entity.Pet;
+import com.savethepet.model.entity.Status;
+import com.savethepet.model.entity.User;
```
Useless

BrineFast commented on be/src/main/java/com/savethepet/model/entity/Pet.java:
```diff
@@ -17,7 +17,18 @@
+
+    private String img;
+
+    private transient PetLocation location;
```
So we can have empty fields?

BrineFast commented on be/src/main/java/com/savethepet/service/PetPageService.java:
```diff
@@ -0,0 +1,117 @@
+                new UserNotFoundException("User with id=" + id.toString() + notFound));
+        return petRepo.findAll(breed, status, id).stream()
+                .map(PetInfoDTO::getDtoFromPet).collect(Collectors.toList());
+    }
```
Why are there so many search? Just search by `user_id`

PaolkaT committed:
TASK 18: Fix shortcomings
